### PR TITLE
g2: updated for 3.4.8 release

### PR DESCRIPTION
**Pull request by edwardhartnett**
Updated package build for 2.4.8 release.

### CODE REVIEW

File: var/spack/repos/builtin/packages/g2/package.py [hunk 1]
```diff
@@ -62,3 +64,8 @@ def setup_run_environment(self, env):
             lib = find_libraries("libg2_" + suffix, root=self.prefix, shared=False, recursive=True)
             env.set("G2_LIB" + suffix, lib[0])
             env.set("G2_INC" + suffix, join_path(self.prefix, "include_" + suffix))
+
+    def check(self):
```
Comment (edwardhartnett) [inline var/spack/repos/builtin/packages/g2/package.py:68]: @AlexanderRichert-NOAA this is in our private version of the package file, should it be only there? Or should this check() function be added to the main spack repo as well?

Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/g2/package.py:68]: I've been including that in our other recipes, so I would say let's keep it here. In principle it should be just as useful to external developers who want to run unit tests.

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256, rest LGTM.

**Comment by tldahlgren** [comment]
The package successfully builds and build-time tests pass for me.

**Comment by tldahlgren** [comment]
Shall I merge the PR or wait for more reviews?

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Pull request merged**